CMakeLists.txt

```diff
@@ -37,14 +37,25 @@ set(CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_SOURCE_DIR}/external/cmake-modules")
 
+option(USE_STRICT_COMPILER_WARNINGS "Use a stricter compiler-warning level" OFF)
+
+if (USE_STRICT_COMPILER_WARNINGS)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wconversion -Wextra -Werror -Wno-unused-parameter -Wno-unused-function -Wno-missing-field-initializers")
```

Comment:
Is `-Werror=shadow` covered from these?

Reply:
but weird, I can not spot `-Wshadow` in any of the lists for `-Wall`, `-Wextra` or `-Wpedantic` - https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Reply:
but I can confirm I'm seeing `-Wshadow` is being enforced (using gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0):
e.g. `error: unused variable ‘valid’ [-Werror=unused-variable]`

Reply:
ok, my bad. I only saw clang-tidy warnings. will include `-Wshadow` and iterate again

Reply:
ok, added `-Wshadow` and fixed all occurances in gfxreconformat/util/graphics. thanks @panos-lunarg !

Reply:
@panos-lunarg , okay with this conversation and can this be resolved?

Reply:
yes, forgot to resolve this